cmd/kots/cli/run.go

```diff
@@ -0,0 +1,111 @@
+// +build kots_experimental
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/replicatedhq/kots/pkg/apiserver"
+	"github.com/replicatedhq/kots/pkg/cluster"
+	"github.com/replicatedhq/kots/pkg/filestore"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/util"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func RunCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "run [upstream uri]",
+		Short:         "Runs an application in an embedded cluster",
+		Long:          ``,
+		SilenceUsage:  true,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			v := viper.GetViper()
+
+			if len(args) == 0 {
+				cmd.Help()
+				os.Exit(1)
+			}
+
+			slug := args[0]
+			log := logger.NewCLILogger()
+			log.Info("Running application %s", slug)
+
+			loggerCtx := context.WithValue(context.Background(), "log", log)
```

Comment:
im not sure i agree with this pattern. im not sure context is the best place for DI